tests/test_builders/test_build_linkcheck.py

```diff
@@ -1439,3 +1439,119 @@ def test_linkcheck_exclude_documents(app: SphinxTestApp) -> None:
         'uri': 'https://www.sphinx-doc.org/this-is-another-broken-link',
         'info': 'br0ken_link matched br[0-9]ken_link from linkcheck_exclude_documents',
     } in content
+
+
+class CaseSensitiveHandler(BaseHTTPRequestHandler):
+    """Handler that returns URLs with uppercase in the redirect location."""
+
+    protocol_version = 'HTTP/1.1'
+
+    def do_HEAD(self):
+        # Simulate a server that returns URLs with different case
+        if self.path == '/path':
+            # Return the path with uppercase
+            self.send_response(200, 'OK')
+            # Simulate the response URL being in uppercase
+            self.send_header('Content-Length', '0')
+            self.end_headers()
+        elif self.path == '/anchor.html':
+            self.send_response(200, 'OK')
+            self.send_header('Content-Length', '0')
+            self.end_headers()
```

Comment:
```suggestion
        elif self.path == '/anchor.html':
            self.send_response(200, 'OK')
            self.send_header('Content-Length', '0')
            self.end_headers()
```

Reply:
I'd recommend removing most/all unused code paths from the test server -- so, for example, this `anchor.html` path can be removed, I think.

Reply:
Fixed!

Reply:
Thank you!  Re-reviewing now...